src/Http/Http.Results/src/ResultsCache.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http;
+
+internal static partial class ResultsCache
+{
+    public static NotFoundObjectHttpResult NotFound { get; } = new(null);
```

Comment:
Out of curiosity, why not align this with the other status code results? You generally don't need to be concerned with back-compat of implementation particularly when all of it was previously non-public.

Reply:
> why not align this with the other status code results?

I did not follow, sorry, what do you mean about it?

Reply:
@pranavkm As of right now, we've decided to avoid a type hierarchy. And I don't think we want `NotFound(object? value = null)` returning a `StatusCodeHttpResult` in some cases and a `NotFoundObjectHttpResult` in others.